sql/src/main/java/io/crate/operation/collect/collectors/LuceneBatchIterator.java

```diff
@@ -142,6 +137,13 @@ private boolean belowMinScore(Scorer currentScorer) throws IOException {
     }
 
     private boolean tryAdvanceDocIdSetIterator() throws IOException {
+        if (weight == null) {
```

Comment:
Wouldn't it make sense to do this in moveNext instead of a potentially inner loop?